File: REST_API_Microservices-master/frontend-react/src/services/MenuService.js
import axios from 'axios';

const MENU_API_BASE_URL = "http://localhost:8082/api/v1/menus";

class MenuService {

    getMenuItems(){
        return axios.get(MENU_API_BASE_URL);
    }

    createMenu(menu){
        return axios.post(MENU_API_BASE_URL, menu);
    }

    getMenuById(menuId){
        return axios.get(MENU_API_BASE_URL + '/' + menuId);
    }

    updateMenu(menu, menuId){
        return axios.put(MENU_API_BASE_URL + '/' + menuId, menu);
    }

    deleteMenu(menuId){
        return axios.delete(MENU_API_BASE_URL + '/' + menuId);
    }
}

export default new MenuService()

File: REST_API_Microservices-master/backend-microservices-rest-api/food-ordering-menu/src/main/java/foodordering/foodorderingmenu/service/MenuService.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package foodordering.foodorderingmenu.service;

import foodordering.foodorderingmenu.data.Menu;
import foodordering.foodorderingmenu.data.MenuRepository;
import java.util.List;
import java.util.Optional;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.PathVariable;

/**
 *
 * @author Nushan
 */
@Service
public class MenuService {
    @Autowired
    private MenuRepository menuRepository;
    
    public Menu createMenu (Menu mnu){
        return menuRepository.save(mnu);
    }
    
    public List<Menu> getAllMenus(){
       List<Menu> mnu = menuRepository.findAll();
       return mnu;
    }
    
    public Menu getMenuById(int id){
        Optional<Menu> mnu = menuRepository.findById(id);
        return mnu.get();       
    }
    public Menu updateMenu (Menu mnu){
        if (getMenuById(mnu.getId())!=null){
            return menuRepository.save(mnu);
        }
        else {
            return createMenu(mnu);
        } 
    }
    
     public void deleteMenu (Menu mnu){
            menuRepository.delete(mnu);
    }
    
}
